addons/controllers/antrea/antreaconfig_util.go

```diff
@@ -295,6 +296,39 @@ func mapAntreaConfigSpec(cluster *clusterv1beta1.Cluster, config *cniv1alpha2.An
 	if semver.Compare(version, "v1.9.0") >= 0 {
 		configSpec.Antrea.AntreaConfigDataValue.FeatureGates.TopologyAwareHints = &config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.TopologyAwareHints
 	}
+	// NSX related
+	if semver.Compare(version, "1.9.0") >= 0 && config.Spec.AntreaNsx.Enable {
+		configSpec.AntreaNsx.Enable = config.Spec.AntreaNsx.Enable
+		if config.Spec.AntreaNsx.BootstrapFrom.Inline != nil {
+			configSpec.AntreaNsx.BootstrapFrom.Inline.NsxManagers = config.Spec.AntreaNsx.BootstrapFrom.Inline.NsxManagers
+			configSpec.AntreaNsx.BootstrapFrom.Inline.ClusterName = config.Spec.AntreaNsx.BootstrapFrom.Inline.ClusterName
+			// NSX cert
+			secret := &corev1.Secret{}
+			err = client.Get(context.TODO(), types.NamespacedName{
+				Namespace: config.Namespace,
+				Name:      config.Name,
+			}, secret)
+			if err != nil {
+				return configSpec, err
+			}
+			if secret.Data == nil {
+				return configSpec, fmt.Errorf("missing secret data")
+			}
+			if _, ok := secret.Data["tls.crt"]; !ok {
+				return configSpec, fmt.Errorf("missing tls.crt")
+			}
+			configSpec.AntreaNsx.BootstrapFrom.Inline.NsxCertRef.TLSCert = string(secret.Data["tls.crt"])
+			if _, ok := secret.Data["tls.key"]; !ok {
+				return configSpec, fmt.Errorf("missing tls.key")
+			}
+			configSpec.AntreaNsx.BootstrapFrom.Inline.NsxCertRef.TLSKey = string(secret.Data["tls.key"])
+		} else if config.Spec.AntreaNsx.BootstrapFrom.ProviderRef != nil {
+			configSpec.AntreaNsx.BootstrapFrom.ProviderRef.ApiVersion = config.Spec.AntreaNsx.BootstrapFrom.ProviderRef.ApiGroup
+			configSpec.AntreaNsx.BootstrapFrom.ProviderRef.Kind = config.Spec.AntreaNsx.BootstrapFrom.ProviderRef.Kind
+			configSpec.AntreaNsx.BootstrapFrom.ProviderRef.Name = config.Spec.AntreaNsx.BootstrapFrom.ProviderRef.Name
+		}
+		configSpec.AntreaNsx.AntreaNsxConfig.InfraType = config.Spec.AntreaNsx.AntreaNsxConfig.InfraType
```

Comment:
I think it's better not to allow customer to define InfraType. InfraType is vSphere, AWS, Azure. It's better for Antrea addon controller to derive the InfraType automatically based on the cluster definition.  For example, if the cluster is deployed on AWS, there must be a AWSMachineTemplate referenced in the ClusterClass spec. I think this detection is a bit more complicated, we can put it in a follow-up patch.

Reply:
just delete this field and use the default values in this version?